controller/api/destination/endpoint_listener.go

```diff
@@ -131,55 +139,57 @@ func (l *endpointListener) SetServiceID(id *serviceID) {
 	}
 }
 
+// Update is called with lists of newly added and/or removed pods in a service
+// and address updates on the listener's gRPC response stream.
+//
+// N.B. that pod is nil on remove addresses.
+//
+// TODO change the type signature to use more precise types.
 func (l *endpointListener) Update(add, remove []*updateAddress) {
-	l.log.Debugf("Update(%+v, %+v)", add, remove)
+	l.log.Debugf("Update: add=%d; remove=%d", len(add), len(remove))
 
 	if len(add) > 0 {
-		update := &pb.Update{
-			Update: &pb.Update_Add{
-				Add: l.toWeightedAddrSet(add),
-			},
+		// If pods were added, send the list of metadata-rich WeightedAddr endpoints.
+		set := &pb.WeightedAddrSet{MetricLabels: l.labels}
+		for _, a := range add {
+			w := l.toWeightedAddr(a)
+			l.log.Debugf("Update: add: addr=%s; pod=%s; %+v", a.Address(), a.Name(), w)
```

Comment:
why give up logging the namespace?

Reply:
hm? the namespace is part of the logging scope. the controller namespace is a process-global thing.

Reply:
also, most of the germaine details are in the weighted addr body at the end

Reply:
`msg="Update: add: addr=10.1.1.160; pod=linkerd-prometheus-7bbc899687-nd9zt; addr:<ip:<ipv4:167838112 > port:9090 > weight:1 metric_labels:<key:\"control_plane_ns\" value:\"linkerd\" > metric_labels:<key:\"deployment\" value:\"linkerd-prometheus\" > metric_labels:<key:\"pod\" value:\"linkerd-prometheus-7bbc899687-nd9zt\" > metric_labels:<key:\"pod_template_hash\" value:\"7bbc899687\" > protocol_hint:<h2:<> > " component=endpoint-listener namespace=linkerd service=linkerd-prometheus`

Reply:
ah, i forgot the pod namespace was part of the logging scope.